plugins/modules/ndo_pod_profile.py

```diff
@@ -0,0 +1,357 @@
+            ops.append(dict(op="add", path=pod_profile_path, value=mso.sent))
+
+    elif state == "absent":
+        if match:
+            ops.append(dict(op="remove", path=pod_profile_path))
+
+    if not module.check_mode and ops:
+        response = mso.request(mso_template.template_path, method="PATCH", data=ops)
+        match = mso_template.get_pod_profile_object(uuid, name, search_object=response)
+        if match:
+            set_pod_profile_policy_names(mso, mso_templates, match.details)
+            mso.existing = match.details  # When the state is present
+        else:
+            mso.existing = {}  # When the state is absent
+    elif module.check_mode and state != "query":  # When the state is present/absent with check mode
+        set_pod_profile_policy_names(mso, mso_templates, mso.proposed)
+        mso.existing = mso.proposed if state == "present" else {}
+    mso.exit_json()
+
+
+def set_pod_profile_policy_names(mso, mso_templates, pod_profile):
+
+    # TODO: leverage update_config_with_template_and_references when podPolicyGroup type is supported by objects API endpoint
+    # "mso/api/v1/templates/objects?type=podPolicyGroup&uuid=0809339e-65f7-4000-a23a-a39241865db8"
+    # {
+    #   "code": 400,
+    #   "message": "unsupported policy type: podPolicyGroup"
+    # }
+    # reference_dict = {
+    #     "policy": {
+    #         "name": "pod_settings",
+    #         "reference": "policy",
+    #         "type": "podPolicyGroup",
+    #     }
+    # }
+    # if pod_profile and pod_profile.get("policy"):
+    #     mso_template.update_config_with_template_and_references(pod_profile, reference_dict)
+
+    # Workaround template looping until podPolicyGroup type is supported by objects API endpoint
+    # Check if a policy is set in the pod profile, this is the UUID of the pod settings policy
+    if pod_profile.get("policy"):
+        pod_settings_uuid = pod_profile.get("policy")
+
+        # Check if the pod settings UUID is already in the cache
+        if pod_settings_uuid in POD_SETTINGS_CACHE:
+            pod_profile["policyName"] = POD_SETTINGS_CACHE[pod_settings_uuid]
+            return pod_profile
+
+        # Retrieve a summary of all fabric_policy templates from NDO and loop through all templates to find the pod settings policy
+        for template in MSOTemplate(mso, "fabric_policy").template:
+            for policy in template.get("policies", []):
+                # Only retrieve the template details if the fabric_policy template has a pod settings policy defined
+                if policy.get("objType") == "podPolicyGroup" and policy.get("count") > 0:
+                    pod_settings_template = mso_templates.get_template("fabric_policy", template.get("templateName"), template.get("templateId"))
+                    match = pod_settings_template.get_pod_settings_object(pod_settings_uuid)
+                    if match:
+                        pod_profile["policyName"] = match.details.get("name")
```

Comment:
Should the pod settings templateId and templateName be added?

Reply:
will add